src/epcpm/staticmodbustoc.py

```diff
@@ -0,0 +1,136 @@
+        with self.h_path.open("w", newline="\n") as h_file:
+            h_file.write("#ifndef __STATICMODBUS_INTERFACE_GEN_H__\n")
+            h_file.write("#define __STATICMODBUS_INTERFACE_GEN_H__\n\n")
+            h_file.write('#include "interface.h"\n\n')
+            h_file.write(
+                "extern InterfaceItem_void * const staticmodbusAddrRegMap[];\n"
+            )
+            h_file.write("\n#endif //__STATICMODBUS_INTERFACE_GEN_H__\n")
+
+        with self.c_path.open("w", newline="\n") as c_file:
+            c_file.write('#include "staticmodbusInterfaceGen.h"\n')
+            c_file.write('#include "interfaceGen.h"\n\n')
+            c_file.write("InterfaceItem_void * const staticmodbusAddrRegMap[] = \n{\n")
+
+            current_addr = 0
+            for member in self.wrapped.children:
+                builder = builders.wrap(
+                    wrapped=member,
+                    parameter_uuid_finder=self.parameter_uuid_finder,
+                    skip_sunspec=self.skip_sunspec,
+                )
+
+                addr_val, c_line = builder.gen()
+
+                # Fill the undefined registers to return NULL.
+                while current_addr < addr_val:
+                    null_line = f"    [{current_addr}] = NULL,\n"
+                    c_file.write(null_line)
+                    current_addr += 1
+                # Write the defined register that returns interfaceItem_<UUID>.
+                c_file.write(c_line)
+                # Update the current address for the next undefined register fill.
+                current_addr = addr_val + 1
+
+            c_file.write("};\n\n")
+
+
+@builders(epcpm.staticmodbusmodel.FunctionData)
+@attr.s
+class FunctionData:
+    wrapped = attr.ib()
+    parameter_uuid_finder = attr.ib()
+    skip_sunspec = attr.ib(default=False)
+
+    def gen(self):
+        uses_interface_item = False
+        if self.wrapped.parameter_uuid is not None:
+            parameter = self.parameter_uuid_finder(self.wrapped.parameter_uuid)
+            uses_interface_item = (
+                isinstance(parameter, epyqlib.pm.parametermodel.Parameter)
+                and parameter.uses_interface_item()
+            )
+
+        type_node = self.parameter_uuid_finder(self.wrapped.type_uuid)
+
+        # Generate the defined register that returns interfaceItem_<UUID> (or NULL for some cases).
+        if (
+            not self.skip_sunspec
+            and uses_interface_item
+            and not self.wrapped.not_implemented
+            and type_node is not None
+            and type_node.name != "staticmodbussf"
+        ):
+            # TODO: CAMPid 9685439641536675431653179671436
+            parameter_uuid = str(self.wrapped.parameter_uuid).replace("-", "_")
+            uuid_interface_val = f"&interfaceItem_{parameter_uuid}"
+        else:
+            uuid_interface_val = "NULL"
+        addr_val = self.wrapped.address
+        c_line = f"    [{addr_val}] = {uuid_interface_val},\n"
```

Comment:
If I remember correctly, we coded the processing code to expect multi-register entries to have their interface item referenced from each of the relevant registers.  This is only filling out the single base register for each item.

From model 17.
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<html>
<body>


Field Type | Applicable Point | Address Offset | Block Offset | Size | Name | Label | Value | Type | Units | SF | R/W | Mandatory M/O | Description | Notes | Modbus Address
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
Fixed Block |   | 6 | 4 | 2 | Rte | Rate |   | uint32 | bps |   | RW | M | Interface baud rate in bits per second | <uuid:eb3b719e-0bce-4665-a035-735ffdd9287f> | 76
Fixed Block |   | 8 | 6 | 1 | Bits | Bits |   | uint16 |   |   | RW | M | Number of data bits per character | <uuid:589123f9-506e-4202-80b9-86c6d6d78d93> | 78


</body>

</html>

I expect `[77]` to be a pointer to the same interface item.

```c
    [76] = &interfaceItem_eb3b719e_0bce_4665_a035_735ffdd9287f,
    [77] = NULL,
```

This is just a quick guess of what this could look like.  Other code would have to be updated accordingly to handle the list of lines rather than the present single line.

```suggestion
        c_lines = [
        	f"[{addr_val + offset}] = {uuid_interface_val},"
        	for offset in range(self.wrapped.size)
      	]
```

This suggestion does integrate a couple tweaks to be more like the other generators that use [`epcpm.c.format_nested_lists()`](https://github.com/altendky/pm/blob/8dcce3f2575f29cca1aebd9956f3e8f6fb4459ad/src/epcpm/c.py#L6-L22) at the end.  Such as in `sunspectobitfieldsc.py` below.  I initially tried using https://github.com/eliben/pycparser to define the C code and format it but that ended up _insanely_ verbose and incomprehensible to non-compiler human brains.  So, I settled on a simple nested list of strings.  Each layer of nesting indicates indentation.  This avoids each layer having to be aware of how nested it is.  It needs only indicate any additional nesting it wants.

https://github.com/altendky/pm/blob/8dcce3f2575f29cca1aebd9956f3e8f6fb4459ad/src/epcpm/sunspectobitfieldsc.py#L93-L94

Reply:
Yes, this was overlooked. I've fixed (based on your recommendations) in fab4ec73e2ee37e5d93099427302e250d3d81abf.